### PR TITLE
Variables invalid at compute time should have the value of "unset"

### LINKED ISSUE #3881: CSS variables without a fallback should use unset as the fallback
### Description

CSS variables that have not be defined should be treated as unset. They are currently ignored. See: https://drafts.csswg.org/css-variables/#invalid-at-computed-value-time

This behavior causes problems when using `useNativeCSSProperties` since the behavior differs between browsers that do and do not support custom properties.
#### Live Demo

https://jsbin.com/viyarefiqo/1/edit?html,output
#### Steps to Reproduce
1. Give an element `color: red`
2. Give the element `color: var(--my-nonexistent-color)` in a rule with more specificity or that appears after in the stylesheet
#### Expected Results

The element inherits the color from the parent
#### Actual Results

The element uses `color: red`
### Browsers Affected

- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari 9
- [x] Safari 8
- [x] IE 11
### Versions
- Polymer: v1.6.1
- webcomponents-lite: 0.7.22

### PR DESCRIPTION
**Pull request by azakus**
Fixes #3881

### CODE REVIEW

File: src/lib/style-properties.html [hunk 1]
```diff
@@ -158,12 +158,15 @@
...
+              if (!value) {
+                return prefix + suffix;
+              }
               var propertyValue = self.valueForProperty(props[value], props);
               // if value is "initial", then the variable should be treated as unset
               if (!propertyValue || propertyValue === 'initial') {
                 // fallback may be --a or var(--a) or literal
                 propertyValue = self.valueForProperty(props[fallback] || fallback, props) ||
-                fallback;
+                fallback || 'unset';
```
Comment (sorvell) [inline src/lib/style-properties.html]: For browsers that do not support `unset` don't we need to place `inherit` or `initial` explicitly?

Comment (azakus) [inline src/lib/style-properties.html]: That would require knowing which properties are `inherit` and `initial` by default, including all shorthands.
Should we bother?

Comment (sorvell) [inline src/lib/style-properties.html]: This seems no worse than before so let's defer that until needed.

**Comment by sorvell** [comment]
LGTM after we get the tests passing...

**Comment by azakus** [comment]
Looks like older browsers like Safari 7 don't handle `unset` like I thought :(

**Comment by azakus** [comment]
This is a dead-end for 1.x because we have moved on. This is implemented in webcomponents/shadycss for 2.x

**Pull request closed without merging**